---
title: "032720_RNAseq_analysis_PART3"
output: 
  html_document:
    keep_md: true
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, error = FALSE, message = FALSE)
```

## Purpose

The goal of this section is to predict which transcription factors may regulate Card9-dependent and Card9-independent Dectin-1 regulated genes. To do this, we are using ImmGen ATAC-seq data to identify open chromatin regions in our cell type of interest (BM neutrophils) and the predicted transcription factor (TF) binding sites in these open regions. This information is then used with a pathway enrichment analysis approach to identify those predicted TF sites that are overrepresented near genes of one group or the other. 

```{r library, results='hide'}
library(devtools)
library(Biobase)
library(goseq)
library(DESeq2)
library(org.Hs.eg.db)
library(org.Mm.eg.db)
library(geneLenDataBase)
library(tidyr)
library(dplyr)
library(readr)
library(VennDiagram)
library(ggplot2)
library(ggrepel)
library(plotly)
```

## 1. Compile ATACseq OCR data from ImmGen files

Processed data files were downloaded from ImmGen for BM neutrophils for each chromosome. Here we are compiling these separate chr files into a single dataframe.

```{r}
#loaded all csv files into a single file
files <- list.files(path = "Immgen_ATACseq_chromosomes", pattern = "*.csv", full.names = T)
tbl <- sapply(files, read_csv, simplify=FALSE) %>% 
bind_rows(.id = "id")

#493013 rows, 17 columns
dim(tbl)

#modifying name, class, factors
tbl$Chr <- as.factor(tbl$Chr)
nlevels(tbl$Chr)
colnames(tbl)
tbl$Genes.within.100kb <- tbl$`Genes within 100Kb`
tbl <- as.data.frame(tbl)
class(tbl)
colnames(tbl)
```

## 2. Generate a tidy dataset, first subsetting on OCRs with values >10

```{r}
#CUTOFF - Remove rows where GN_BM is <10
OCR_threshhold <- 10
tbl_cutoff <- subset(tbl, tbl[,'GN_BM'] > OCR_threshhold)  

#38997 rows, 18 columns
dim(tbl_cutoff)
```

Gene names into rows

```{r}
#making tidy - splitting comma separated gene names into rows - took ~2min to run
start_time1 <- Sys.time()
tbl_1 <- tbl_cutoff %>% 
    mutate(Genes.within.100kb = strsplit(as.character(Genes.within.100kb), ",")) %>% 
    unnest(Genes.within.100kb)
end_time1 <- Sys.time()
end_time1 - start_time1
dim(tbl_1)
```

TF names into rows

```{r}
#making tidy - splitting transcription factor names into rows
#try replacing | with space
tbl_2 <- gsub("\\|", " ", tbl_1$`TF Name`)
class(tbl_2)
#created character vector now add as new column to original dataframe
tbl_1$TF_name_space <- tbl_2


#separating each letter instead of each TF name.... took a very long time to run
start_time2 <- Sys.time()
tbl_3 <- tbl_1 %>% 
    mutate(TF_name_space = strsplit(as.character(TF_name_space), " ")) %>% 
    unnest(TF_name_space)
end_time2 <- Sys.time()
end_time2 - start_time2
```

Looking at features of and saving the compiled file

```{r}
#now ~5 million rows long (with tidy gene + tf combinations)
dim(tbl_3)
head(tbl_3)
colnames(tbl_3)

#MAKE my columns of interest factors
tbl_3$TF_name_space <- as.factor(tbl_3$TF_name_space)
tbl_3$Genes.within.100kb <- as.factor(tbl_3$Genes.within.100kb)
#number of levels: 1205 TFs, 18071 genes
nlevels(tbl_3$TF_name_space)
nlevels(tbl_3$Genes.within.100kb)

#Save file
start_time3 <- Sys.time()
save(tbl_3, file = "Immgen_ATACSeq_GNBM_OCR_Compiled_GN10cutoff.Rdata")
end_time3 <- Sys.time()
end_time3 - start_time3
```

## SessionInfo

```{r}
sessionInfo()
```

